### PR TITLE
Sync GHA workflow w/ the collection template

### PR DESCRIPTION
**Pull request by webknjaz**
##### SUMMARY

$sbj

##### ISSUE TYPE

- Maintenance Request
- Testing Pull Request

##### COMPONENT NAME

GHA

##### ADDITIONAL INFORMATION

N/A

### CODE REVIEW

File: .github/workflows/ansible-test-plugins.yml [hunk 1]
```diff
@@ -30,38 +30,12 @@ jobs:
           - stable-2.13
           - stable-2.14
           - devel
-        python:
-          - 3.8
-          - 3.9
```
Comment (webknjaz) [inline .github/workflows/ansible-test-plugins.yml:35]: This was never necessary w/ `--docker`.

File: .github/workflows/ansible-test-plugins.yml [hunk 2]
```diff
@@ -117,51 +91,29 @@ jobs:
...
-        with:
-          fail_ci_if_error: false
+          ansible-core-version: ${{ matrix.ansible }}
+          pre-test-cmd: >-
+            DB_ENGINE=$(echo '${{ matrix.db_engine_version }}' | awk -F_ '{print $1}');
+            DB_VERSION=$(echo '${{ matrix.db_engine_version }}' | awk -F_ '{print $2}');
+            DB_ENGINE_PRETTY=$([[ "${DB_ENGINE}" == 'mysql' ]] && echo 'MySQL' || echo 'MariaDB');
+            >&2 echo Matrix factor for the DB is ${{ matrix.db_engine_version }}...;
+            >&2 echo Setting ${DB_ENGINE_PRETTY} version to ${DB_VERSION}...;
+            sed -i "s/^${DB_ENGINE}_version:.*/${DB_ENGINE}_version: $DB_VERSION/g" -e 's/^mariadb_install: false/mariadb_install: true/g' '${{ env.mysql_version_file }}';
```
Comment (webknjaz) [inline .github/workflows/ansible-test-plugins.yml]:
```suggestion
            sed -i -e "s/^${DB_ENGINE}_version:.*/${DB_ENGINE}_version: $DB_VERSION/g" -e 's/^mariadb_install: false/mariadb_install: true/g' '${{ env.mysql_version_file }}';
```

File: .github/workflows/ansible-test-plugins.yml [hunk 3]
```diff
@@ -117,51 +91,29 @@ jobs:
...
+          pre-test-cmd: >-
+            DB_ENGINE=$(echo '${{ matrix.db_engine_version }}' | awk -F_ '{print $1}');
+            DB_VERSION=$(echo '${{ matrix.db_engine_version }}' | awk -F_ '{print $2}');
+            DB_ENGINE_PRETTY=$([[ "${DB_ENGINE}" == 'mysql' ]] && echo 'MySQL' || echo 'MariaDB');
+            >&2 echo Matrix factor for the DB is ${{ matrix.db_engine_version }}...;
+            >&2 echo Setting ${DB_ENGINE_PRETTY} version to ${DB_VERSION}...;
+            sed -i -e "s/^${DB_ENGINE}_version:.*/${DB_ENGINE}_version: $DB_VERSION/g" -e 's/^mariadb_install: false/mariadb_install: true/g' '${{ env.mysql_version_file }}';
+            ${{
+              matrix.db_engine_version == 'mariadb_10.8.3'
+              && '>&2 echo Set MariaDB v10.8.3 URL sub dir...; sed -i -e "s/^mariadb_url_subdir:.*/mariadb_url_subdir: linux-systemd/g" "${{ env.connector_version_file }}";'
```
Comment (webknjaz) [inline .github/workflows/ansible-test-plugins.yml]:
```suggestion
              && '>&2 echo Set MariaDB v10.8.3 URL sub dir...; sed -i -e "s/^mariadb_url_subdir:.*/mariadb_url_subdir: linux-systemd/g" ${{ env.connector_version_file }};'
```

File: .github/workflows/ansible-test-plugins.yml [hunk 4]
```diff
@@ -117,51 +91,29 @@ jobs:
...
+          pre-test-cmd: >-
+            DB_ENGINE=$(echo '${{ matrix.db_engine_version }}' | awk -F_ '{print $1}');
+            DB_VERSION=$(echo '${{ matrix.db_engine_version }}' | awk -F_ '{print $2}');
+            DB_ENGINE_PRETTY=$([[ "${DB_ENGINE}" == 'mysql' ]] && echo 'MySQL' || echo 'MariaDB');
+            >&2 echo Matrix factor for the DB is ${{ matrix.db_engine_version }}...;
+            >&2 echo Setting ${DB_ENGINE_PRETTY} version to ${DB_VERSION}...;
+            sed -i -e "s/^${DB_ENGINE}_version:.*/${DB_ENGINE}_version: $DB_VERSION/g" -e 's/^mariadb_install: false/mariadb_install: true/g' '${{ env.mysql_version_file }}';
+            ${{
+              matrix.db_engine_version == 'mariadb_10.8.3'
+              && '>&2 echo Set MariaDB v10.8.3 URL sub dir...; sed -i -e "s/^mariadb_url_subdir:.*/mariadb_url_subdir: linux-systemd/g" ${{ env.connector_version_file }};'
```
Comment (webknjaz) [inline .github/workflows/ansible-test-plugins.yml]:
```suggestion
              && format(
                '>&2 echo Set MariaDB v10.8.3 URL sub dir...; sed -i -e "s/^mariadb_url_subdir:.*/mariadb_url_subdir: linux-systemd/g" "{0}";', env.connector_version_file
              )
```

**Comment by Andersson007** [comment]
@webknjaz thanks for doing this!
cc @laurent-indermuehle @rsicart

**Comment by laurent-indermuehle** [comment]
@webknjaz thanks for the clean up!

I've a question about the task "Check out code". Is it really unnecessary? If yes, we could remove it from the roles' tests.

**Comment by webknjaz** [comment]
@laurent-indermuehle these tasks are removed in the workflow where ansible-test is used. But I didn't touch the other workflow that is supposed to run molecule. It is out of the scope in this PR so I didn't even look at it. Now that I did, I think the whole workflow can be removed as it's almost a no-op — molecule invocation is commented out and only the dependencies are installed there. Still, the goal of this PR is to replace manual ansible-test invocation with an action which it does. I'm not going to change other workflows here.

**Comment by laurent-indermuehle** [comment]
@webknjaz the role workflow is there in case this collection hosts one some day. As you explained, it uses molecule and not ansible-test, so no need for change it.

Now I get what you did! Thanks for the work and the explanations! LGTM.

**Comment by Andersson007** [comment]
@laurent-indermuehle if you're OK with the changes, feel free to merge and backport the PR

**Pull request merged**